File: CS315-HW2/Codes/altiparmak_gokberk.lua
-- Conditional exit
for i= 3,1,-1
do
    print("*")
end

-- Unconditional exit(break)
x = 4
repeat
print("*")
break;
until( x == 3 )

-- Unconditional exit(continue)
repeat
    x = x + 1;
    if (x < 10) then
        goto continue
    end
    print("*")
    break
    ::continue::
until( x == 3 )

-- Enclosed loops exit
for i = 0,2,1
do
    for j = 0,1,1
    do 
        if (i == 1) then
            break;
        end
        print("j = " , j) 
    end
        print("i = " , i) 
end

-- Labeled exit
for i = 5,1,-1
do
    for j = 3,1,-1
    do 
        goto exit
    end
end
::exit::
print("***")

File: CS315-HW2/Codes/altiparmak_gokberk.dart
void main() {
    // Conditional exit
    for (int i = 0; i < 5; i++) {
        print("*");
    }

    // Unconditional exit(break)
    var x = 4;
    for (;;) {
        if (x == 5)
            break;
        print("*")
        x++;
    }

    // Unconditional exit(continue)
    for(;;) {
        x++;
        if (x < 10)
            continue;
        print("*");
        break;
    }

    // Enclosed loops exit
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 2; j++) {
            if (i == 1) {
                break;
            }
            print('j = $j');
        }
        print('i = $i');
    } 

    // Labeled exit
    outFor:
    for (int i = 0; i < 5; i++) {
        for (int j = 0; j < 3; j++) {
            print("*");
            break outFor;
        }
    }
}